src/TheThingsNetwork.h

```diff
@@ -8,6 +8,9 @@
 #include <Stream.h>
 #include <avr/pgmspace.h>
 
+// If you need to use soft serial library you need to uncomment this line
+//#define USE_SOFT_SERIAL
```

Comment:
We can't expect people to uncomment lines somewhere in libraries. Also, that will be overwritten when updating them.

Reply:
The problem is that we can't do that in sketch, not that I do not want, it won't just work because when the compiler compile the library it does not know anything about what is done in the sketch, nor defined values with `#define`

So a decision is on the go, using stream instead of HardwareSerial makes sense to have majority devices compatible (which are not TTN one) without changing a line of code in the library. But on the other hand it impact the LowPower mode for TTN devices that use all Hardware Serial.

From my point of view TTN lib should match TTN devices and make them working as expected (ie no impact on battery life), and I think it’s what TTN backers and devices buyers want but I may be wrong.

Another vision is to make others boards users happy and be able to use soft serial on other devices (not TTN one) without changing any line of code in the library. This will impact severely battery life for users in the case above that wait for a battery life for one year.

But it's not my decision.

Another way could be having dedicated TTN Boards (As sparkfun, Adafruit, ..) in Arduino IDE, for example `The Things Node` and 'The Things Uno`  and select these board to build and upload. This way, library know the target boards and can do some conditional compilation.

Reply:
Ok, we now have a fix, Samuel created TTN boards and we tested together.

Need to install new TTN boards from Arduino IDE Boards URL. In IDE preference add this additional board manager URL
https://raw.githubusercontent.com/samuel-puschacher/TTP_Arduino_Boards/master/package_thethingsproducts_index.json 

Once added, install from board manager install TheThingsNetwork boards
![image](https://user-images.githubusercontent.com/2471931/38060759-678727a4-32ec-11e8-8f96-0c5996c6bdd1.png)

Once done Select the correct board as target
![image](https://user-images.githubusercontent.com/2471931/38060783-90f3f5a4-32ec-11e8-83f2-d482f380f0b6.png)

If selected The Things Uno or The Things Node, it will use HardwareSerial and LowPowerFix will be active, else, stay as usual Stream.

Hope this is satisfying ?

Reply:
So I hope, we can bring this further now

Reply:
This solution is even downwards compatible ....
And the best the docs are already updated, see PR: https://github.com/TheThingsNetwork/docs/pull/226

Reply:
With in the sketch, I mean in the user sketch, the baud rate thing. But I think a board is a very good suggestion.